src/player/teleportation_stones.js

```diff
@@ -63,6 +63,32 @@ function teleportation_stones_in_chunk(world, chunk_x, chunk_z) {
   )
 }
 
+ function is_player_near(Xplayer,Yplayer, Xstone, Ystone) {
+  const radius = 30
+  if((Math.abs(Xstone-Xplayer) <= radius) && (Math.abs(Ystone-Yplayer) <= radius)) {
+    return true
+  }else{
+    return false
+  }
+}
+
+/**
+ * Return the closest teleportation stone from the player
+ * @param {*} world 
+ * @param {*} player_pos_x 
+ * @param {*} player_pos_y 
+ * @returns {stones} name of the teleportation stone
+ */
+export function closest_stone(world,player_pos_x,player_pos_y) {
+  const stones = world.teleportation_stones
+  const zone = stones
+  .filter(stone => is_player_near(player_pos_x, player_pos_y, stone.position.x, stone.position.y))
+  if(zone.length === 0) {
+    return "wilderness"
+  }
+  return `${zone.map(stone => stone.name)}`
+  }
```

Comment:
That function is ambiguous as it says closest_stone not closest_stones_names, it should return a single stone or have its name changed. But it's better to return a simple stone. (also wilderness is not a stone)

You don't need the world, just the stones array, also it's better to use an object argument, or in the case a signature as below

> A player can't be in 2 different zones btw

> the distance function comes from `src/math.js`

```suggestion
const PLAYER_ZONE_DISTANCE = 30

// note that a player at equal distance of 2 stones
// would be at the zone limit so it doesn't matter if we say one or the other
export function closest_stone(stones, { position }) {
  return stones.find(stone => distance(stone.position, position) <= PLAYER_ZONE_DISTANCE)
}
```

that function returns a stone or `undefined` so you can verify as
`const zone = closest_stone() ?? 'wilderness'`